test/test_binary_ufuncs.py

```diff
@@ -2761,6 +2761,60 @@ def _compare_helper(x, y, torch_fn, reference_fn):
         _compare_helper(t, zeros, *xlog1py_fns)
         _compare_helper(t, 0., *xlog1py_fns)
 
+    @dtypes(*product(torch.testing.get_all_dtypes(include_complex=False,
+                                                  include_half=False, include_bfloat16=False),
+                     torch.testing.get_all_dtypes(include_complex=False,
+                                                  include_half=False, include_bfloat16=False)))
+    @skipIf(not TEST_SCIPY, "Scipy required for the test.")
+    def test_zeta(self, device, dtypes):
+        x_dtype, q_dtype = dtypes
+
+        def test_helper(x, q):
+            x_np = x if isinstance(x, float) else x.cpu().numpy()
+            q_np = q if isinstance(q, float) else q.cpu().numpy()
+            expected = torch.from_numpy(scipy.special.zeta(x_np, q_np))
+            actual = torch.special.zeta(x, q)
+
+            rtol, atol = None, None
+            if self.device_type == 'cpu':
+                rtol, atol = 1e-6, 1e-6
+            self.assertEqual(expected, actual, rtol=rtol, atol=atol, exact_dtype=False)
```

Comment:
Had to add tolerance based on CI failures (for jobs with relatively old compilers).

https://app.circleci.com/pipelines/github/pytorch/pytorch/338469/workflows/db193abd-5b89-405a-9987-f4926107e2ec/jobs/14225591

https://app.circleci.com/pipelines/github/pytorch/pytorch/338469/workflows/db193abd-5b89-405a-9987-f4926107e2ec/jobs/14224886

Reply:
cc @pmeier for yet another example of binary elementwise testing we should automate